controllers/updateservice_controller.go

```diff
@@ -350,6 +351,38 @@ func (r *UpdateServiceReconciler) ensureTrustedClusterCA(ctx context.Context, re
 	return err
 }
 
+func (r *UpdateServiceReconciler) ensureGraphDataSHA(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService,
+	resources *kubeResources) (string, error){
+
+	pod := resources.graphDataPodContainer
+	if err := controllerutil.SetControllerReference(instance, pod, r.Scheme); err != nil {
+		return "", err
+	}
+
+	// Check if this pod already exists
+	found := &corev1.Pod{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+
+	if err != nil && apiErrors.IsNotFound(err) || found.Status.Phase == "Succeeded"{
+		reqLogger.Info("Creating Pod", "Namespace", pod.Namespace, "Name", pod.Name)
+		err := r.Client.Create(ctx, pod)
+		if err != nil {
+			handleErr(reqLogger, &instance.Status, "CreateGraphDataPodFailed", err)
+		}
+		return "", err
+	} else if err != nil {
+		handleErr(reqLogger, &instance.Status, "GetGraphDataPodFailed", err)
+		return "", err
+	}
+
+	if len(found.Status.ContainerStatuses) > 0 {
+		return found.Status.ContainerStatuses[0].ImageID, nil
+	}
+
+	return "", nil
```

Comment:
Eventually in the "we got the pod, but there's no container status yet" case, we might want to bail out and try again with a fresh pod.  Some kind of `status` check to see if the pod is in the "still trying to create containers" state or the "we've given up" state?

Reply:
added